From 3005d497a1b20c976b0359ca159824edb3174c10 Mon Sep 17 00:00:00 2001
From: Tim Clephas <tim.clephas@nobleo.nl>
Date: Mon, 29 Jul 2024 07:36:27 +0200
Subject: [PATCH] Single package.xml file for both ROS1 and ROS2 #48

---
 CMakeLists.txt | 57 +++++++++++++++-----------------------------------
 package.xml    |  8 +++----
 2 files changed, 20 insertions(+), 45 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d28e77..4917155 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,11 +7,6 @@ project(rslidar_sdk)
 #=======================================
 set(POINT_TYPE XYZI)
 
-#=======================================
-# Compile setup (ORIGINAL, CATKIN, COLCON)
-#=======================================
-set(COMPILE_METHOD CATKIN)
-
 option(ENABLE_TRANSFORM "Enable transform functions" OFF)
 if(${ENABLE_TRANSFORM})
   add_definitions("-DENABLE_TRANSFORM")
@@ -63,12 +58,7 @@ if (CMAKE_BUILD_TYPE STREQUAL "")
   add_definitions(-O3)
 endif()
 
-if($ENV{ROS_DISTRO} STREQUAL "humble")  # the ros2 humble requires c++17
 add_definitions(-std=c++17)
-else()
-add_definitions(-std=c++14)
-endif()
-
 add_compile_options(-Wall)
 
 #========================
@@ -103,21 +93,16 @@ if(roscpp_FOUND)
   include_directories(${roscpp_INCLUDE_DIRS} ${roslib_INCLUDE_DIRS})
   set(ROS_LIBS ${roscpp_LIBRARIES} ${roslib_LIBRARIES})
 
-  #Catkin#
-  if(${COMPILE_METHOD} STREQUAL "CATKIN")
-
-    add_definitions(-DRUN_IN_ROS_WORKSPACE)
-
-    find_package(catkin REQUIRED COMPONENTS
-      roscpp
-      sensor_msgs
-      roslib)
+  add_definitions(-DRUN_IN_ROS_WORKSPACE)
 
-    catkin_package(CATKIN_DEPENDS 
-      sensor_msgs 
-      roslib)
+  find_package(catkin REQUIRED COMPONENTS
+    roscpp
+    sensor_msgs
+    roslib)
 
-  endif(${COMPILE_METHOD} STREQUAL "CATKIN")
+  catkin_package(CATKIN_DEPENDS 
+    sensor_msgs 
+    roslib)
 
 else(roscpp_FOUND)
 
@@ -130,8 +115,7 @@ endif(roscpp_FOUND)
 #ROS2#
 find_package(rclcpp QUIET)
 
-if(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
-
+if(rclcpp_FOUND)
   message(=============================================================)
   message("-- ROS2 Found. ROS2 Support is turned On.")
   message(=============================================================)
@@ -145,13 +129,11 @@ if(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
   find_package(rslidar_msg REQUIRED)
   find_package(std_msgs REQUIRED)                      
 
-else(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
-
+else(rclcpp_FOUND)
   message(=============================================================)
   message("-- ROS2 Not Found. ROS2 Support is turned Off.")
   message(=============================================================)
-
-endif(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
+endif(rclcpp_FOUND)
 
 #Others#
 find_package(yaml-cpp REQUIRED)
@@ -182,19 +164,15 @@ if(roscpp_FOUND)
   target_link_libraries(rslidar_sdk_node 
     ${ROS_LIBS})
 
-  if(${COMPILE_METHOD} STREQUAL "CATKIN")
-
-    install(TARGETS rslidar_sdk_node
-            RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-            LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-            ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
-
-  endif()
+  install(TARGETS rslidar_sdk_node
+          RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+          LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+          ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
 
 endif(roscpp_FOUND)
 
 #Ros2#
-if(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
+if(rclcpp_FOUND)
 
   ament_target_dependencies(rslidar_sdk_node 
     rclcpp 
@@ -213,5 +191,4 @@ if(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
 
   ament_package()
 
-endif(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
-
+endif(rclcpp_FOUND)
diff --git a/package.xml b/package.xml
index b7f43f0..e9a9790 100644
--- a/package.xml
+++ b/package.xml
@@ -1,21 +1,19 @@
 <?xml version="1.0"?>
 <package format="3">
   <name>rslidar_sdk</name>
-  <version>1.5.14</version>
+  <version>1.5.0</version>
   <description>The rslidar_sdk package</description>
-  <maintainer email="zdxiao@robosense.cn">robosense</maintainer>
-  <maintainer email="tim.clephas@nobleo.nl">Tim Clephas</maintainer>  <!-- Buildfarm releases -->
+  <maintainer email="ron.zheng@robosense.cn">robosense</maintainer>
   <license>BSD</license>
 
   <buildtool_depend condition="$ROS_VERSION == 1">catkin</buildtool_depend>
   <buildtool_depend condition="$ROS_VERSION == 2">ament_cmake</buildtool_depend>
 
   <depend>libpcap</depend>
-  <depend>libpcl-all-dev</depend>
-  <depend>pcl_conversions</depend>
   <depend condition="$ROS_VERSION == 1">pcl_ros</depend>
   <depend condition="$ROS_VERSION == 2">rclcpp</depend>
   <depend condition="$ROS_VERSION == 1">roscpp</depend>
+  <depend condition="$ROS_VERSION == 1">roslib</depend>
   <depend condition="$ROS_VERSION == 2">rslidar_msg</depend>
   <depend>sensor_msgs</depend>
   <depend>std_msgs</depend>
-- 
2.43.0

