From 8546feb3bae14027f17af1393a17f0e9a3eb2b07 Mon Sep 17 00:00:00 2001
From: Tim Clephas <tim.clephas@nobleo.nl>
Date: Thu, 13 Jan 2022 08:46:47 +0100
Subject: [PATCH] Install package.xml

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64f6acc..045378c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -204,3 +204,6 @@ if(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
   )
   ament_package()
 endif(rclcpp_FOUND AND ${COMPILE_METHOD} STREQUAL "COLCON")
+
+# Install package.xml
+install(FILES package.xml DESTINATION share/rslidar_sdk)
--
2.25.1
